### PR TITLE
Add pendingReason to Subscription

### PR DESCRIPTION
**Pull request by 2e3s**

### CODE REVIEW

**Comment by 2e3s** [comment]
devs please review

File: spec/components/schemas/Subscription.yaml [hunk 1]
```diff
@@ -34,6 +32,23 @@ properties:
...
       - suspended
       - paused
+  pendingReasons:
+    type: array
+    description: |
+      A list of reasons why the subscription should stay in the pending status.
+      - "initial-payment" blocks the activation until the initial invoice is paid
+      - "start-time" blocks the activation until startTime < now
+      - "custom" is to be used when a business requires a later activation.
+        The subscription won't be activated until the field is updated without this pending reason.
```
Comment (adamaltman) [inline spec/components/schemas/Subscription.yaml]: I think we can explain this more clearly.  I think it's the right idea.

**Comment by 2e3s** [comment]
@slavcodev has suggested another approach which I've summarized in #641
@adamaltman please, review this proposal too.

**Comment by 2e3s** [comment]
It has become outdated, the solution should be redefined.

**Pull request closed without merging**